Jenkinsfiles/release-satellite.groovy

```diff
@@ -0,0 +1,164 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Kevin Leturc <kleturc@nuxeo.com>
+ *     Manon Lumeau <mlumeau@nuxeo.com>
+ */
+
+import org.kohsuke.github.GHRepository
+import org.kohsuke.github.GitHubBuilder
+import org.yaml.snakeyaml.DumperOptions
+@Grab("org.yaml:snakeyaml:1.16")
+import org.yaml.snakeyaml.Yaml
+
+properties([
+  [$class: 'BuildDiscarderProperty', strategy: [$class: 'LogRotator', daysToKeepStr: '60', numToKeepStr: '60', artifactNumToKeepStr: '1']],
+  disableConcurrentBuilds(),
+  [$class: 'RebuildSettings', autoRebuild: false, rebuildDisabled: false],
+  parameters([
+    choice(name: 'REPOSITORY', description: 'Doc repository to release', choices: ['doc.nuxeo.com-client-java']),
+    string(name: 'ORIGIN_BRANCH', description: 'Origin Branch', defaultValue: 'master'),
+    string(name: 'VERSION', description: 'New version to release'),
+    string(name: 'TICKET', description: 'Jira ticket to use for commits'),
+    booleanParam(name: 'DRY_RUN', description: 'Try to upgrade the repository without committing', defaultValue: false),
+  ]),
+])
+
+node('SLAVEPRIV') {
+  def docRepositoryName = env.REPOSITORY
+  def originBranch = env.ORIGIN_BRANCH
+  def targetBranch = env.VERSION
+  def targetVersion = env.VERSION
+  def techno = docRepositoryName.replaceFirst('.*-client-(.+)', '$1')
+  def githubCredential = "c203b7ce-203b-4fcb-81eb-8e1cae7cb6c2"
+  def docRepository = getGithubRepository(githubCredential, docRepositoryName)
+  stage('create branch') {
+    createBranch(docRepository, originBranch, targetBranch, env.DRY_RUN)
+
+    updateYamlConfigAndPush(docRepository, targetBranch, "Make ${targetVersion} version live", env.TICKET, env.DRY_RUN) { configYml ->
+      def versions = configYml.default.site.versions
+      def branchVersionIdx = versions.findIndexOf { it.label == "Next Java" }
+      // can't use interpolation here, otherwise snakeyaml will have difficulties
+      versions[branchVersionIdx].label = 'Java ' + targetVersion
+      versions[branchVersionIdx].url_path = targetVersion
+    }
+  }
+  stage('update master') {
+    updateYamlConfigAndPush(docRepository, originBranch, "Add new ${targetVersion} version", env.TICKET, env.DRY_RUN) { configYml ->
+      // version to insert
+      def version = [label: 'Java ' + targetVersion, url_path: targetVersion]
+
+      // look for the ongoing version
+      def versions = configYml.default.site.versions
+      def branchVersionIdx = versions.findIndexOf { it.label == "Next Java" }
+
+      if (branchVersionIdx == versions.size() - 1) {
+        versions.add(version)
+      } else {
+        versions.add(branchVersionIdx + 1, version)
+      }
+      configYml.default.site.versions = versions
+    }
+  }
+  stage('update other branches') {
```

Comment:
It would be nice to create PRs when updating the branches.

Reply:
Not sure for this part as it is technical and it will grows with the number of versions, we currently have 10 versions, which makes a lot of PR.